model/Translation/Service/QtiLanguageSetter.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2024 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\Translation\Service;
+
+use core_kernel_classes_Resource;
+use oat\generis\model\data\Ontology;
+use oat\tao\model\TaoOntology;
+use oat\taoQtiItem\model\qti\Service;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+class QtiLanguageSetter
```

Comment:
- [ ] Can you add a test, please?

Reply:
Done: https://github.com/oat-sa/extension-tao-itemqti/pull/2599/commits/c39098d107ac191ac364b9d387f6985547b6e21d